### DISCUSSION TITLE
Feature Request: Return "Unauthorized" when webhook auth token is illegal (instead of "Not Found" )

### DISCUSSION THREAD

**Discussion by kenchon** [community/community#26368] [Actions]
<p>I was using following webhook api to run GitHub Actions</p>
<ul>
<li><a href="https://docs.github.com/ja/free-pro-team@latest/rest/reference/repos#create-a-repository-dispatch-event" rel="noopener nofollow ugc">Create a repository dispatch event</a></li>
</ul>
<pre><code class="lang-auto">POST /repos/{owner}/{repo}/dispatches
</code></pre>
<p>This API returns <code>"404 Not Found"</code> when Authorization Token was ill-formed. At first, I didn’t understand why I get 404 since specified endpoint was correct. If there’s no specific reasons, I think <code>"401 Unauthorized"</code> is proper message for this case :pleading_face:</p>

**Comment by Simran-B** [comment]
<p>I think this behavior is somewhat consistent, e.g. there is a private repository <a href="https://github.com/github/rest-api-operations" rel="noopener nofollow ugc">https://github.com/github/rest-api-operations</a> but GitHub gives you a 404 if you’re not an authorized member. Could be considered security by obscurity I suppose?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/kenchon">@kenchon</a>,</p>
<p>In most cases, when a user tries to visit a page that he does not have the access on GitHub, he will get the “<strong>404 Not Found</strong>”.<br>
I think we can explain this with the two points below:</p>
<ol>
<li>
<p>GitHub searches the pages within the accessible scopes of the user’s token (or the user). If the requested page is not found within the accessible scopes, the “<strong>404 Not Found</strong>” returned.</p>
</li>
<li>
<p>If the requested page is not found within the accessible scopes of the user, it is completely unnecessary to tell the user that the requested page really exists. As <a class="mention" href="/simran-b">@Simran-B</a> mentioned, it may be a security policy that using an obscure message.</p>
</li>
</ol>

**Comment by kenchon** [comment]
<p>Thanks for your replies.</p>
<p>Take that into consideration, IMO, message like “no resources found under your scope” is more balanced in terms of developer-friendliness and security.</p>
<p>Anyway, your comments are really helpful to understand how 404 comes from (and it is consist behavior through GitHub services).</p>
